### DISCUSSION TITLE
Is There a Package Convert Tailwind Classes into Final Style Properties?

### DISCUSSION THREAD

**Discussion by jon9090** [tailwindlabs/tailwindcss#14108] [Help]
Is there a package that can accept Tailwind classes and output the final style properties?

For example, given this input:

```
!inline-block ltr:text-left rtl:text-right border-border w-[calc(100%)] min-w-0 pointer-events-none truncate
```

It will produce:

```
min-width: 0; 
overflow: hidden;
text-overflow: ellipsis;
width: calc(100%);
...
```

**Comment by philipp-spiess** [comment]
Hey @jon9090! Not that I’m aware of, it's also not going to be easy for more advanced rules (e.g. how should media queries resolve?)

One thing that we do for testing that you can try is to run Tailwind via Puppeteer or Playwright and then use the [`getComputedStyle()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle) DOM API to get the resolved style.

I’m curious what your use case for this is, though 🙂
